cltk/inflection/old_norse/nouns.py

```diff
@@ -1,9 +1,900 @@
-"""Noun declensions"""
+"""Noun declensions
+
+This module intends to
+- set inflected forms of Old Norse nouns,
+- guess declined forms just with the nominative singular, the genitive singular and the nominative plural.
+
+Old Norse nouns vary according to gender (masculine, feminine and neuter), the number (singular and plural) and the case
+(nominative, accusative, dative and genitive). They are classified into the strong and the weak nouns.
+
+All dative plural nouns finish with -um ending.
+All genitive plural nouns finish with -a ending.
+
+Commented doctests do not work as expected, because there is no way, for now, to guess correctly all the forms.
+
+"""
 
 import cltk.inflection.utils as decl_utils
+from cltk.phonology.syllabify import Syllabifier, Syllable
+from cltk.corpus.old_norse.syllabifier import invalid_onsets, BACK_TO_FRONT_VOWELS, VOWELS, CONSONANTS
+from cltk.inflection.old_norse.phonemic_rules import extract_common_stem, apply_u_umlaut, has_u_umlaut
 
 __author__ = ["Clément Besnier <clemsciences@aol.com>", ]
 
 sumar = [["sumar", "sumar", "sumri", "sumars"], ["sumur", "sumur", "sumrum", "sumra"]]
-noun_sumar = decl_utils.DeclinableOneGender("sumar", decl_utils.Gender.neuter)
+noun_sumar = decl_utils.Noun("sumar", decl_utils.Gender.neuter)
 noun_sumar.set_declension(sumar)
+
+s = Syllabifier(language="old_norse", break_geminants=True)
+s.set_invalid_onsets(invalid_onsets)
+
+
+class OldNorseNoun(decl_utils.Noun):
```

Comment:
There are some exceptions like nouns for family members. My main aim is to use the Zoëga's dictionary (available in an other repository of cltk) with this class and make a lemmatizer (here for nouns).

Reply:
> Zoëga's dictionary (available in an other repository of cltk

This is what I suspected. In your current code, it is not reading from this dictionary yet, right?

I think I'd like to see the entire thing "hooked up" from beginning to end. This has two benefits: a) we can understand the entire purpose easier and b) it prevents us from adding potentially useful, but currently unused code (something we usually try to avoid in the master branch).

Reply:
The problem with the complete implementation is that it is too big. The aim is easier to understand but the code would be just too hard to get. Moreover, the small parts may be used for other things.

By the way, I'm not sure that the Old Norse code has ever been used (except me)!